glide-core/src/client/value_conversion.rs

```diff
@@ -522,14 +560,66 @@ fn convert_flat_array_to_array_of_pairs(
     Ok(Value::Array(result))
 }
 
+/// Converts a flattened array with key-values to Maps.  The inner values may be another flattened array with key-values.
```

Comment:
Might want to mention somewhere in here that the innermost field-value pairs are kept in flattened array format instead of a map because they may contain duplicates